[gh project item-list] Add `iterationId` field in ProjectV2ItemFieldIterationValue

iamazeem opened the pull request:
Fixes #10301.

Sample:

```shell
$ bin/gh project item-list 2 --owner openssl --format json --jq '[.items[].iteration // empty][0]'
{
  "duration": 14,
  "iterationId": "5236f443",
  "startDate": "2023-05-18",
  "title": "Iteration 1"
}
```

iamazeem committed:
[gh project item-list] Add `iterationId` field in ProjectV2ItemFieldIterationValue

iamazeem committed:
Merge branch 'trunk' into 10301-gh-project-iteration-id

andyfeller commented:
Thank you for opening this pull request and contributing to the GitHub CLI, @iamazeem! Changes look simple and I've been able to confirm them against an internal project that the iteration info is returned. 👍

While we wait for automated tests to run, my only concern / question is what should be done to the tests in for this change?

https://github.com/cli/cli/blob/4b755bec0484ff756f1eb51e9c0f57b5bd5cbdb3/pkg/cmd/project/shared/queries/queries_test.go

https://github.com/cli/cli/blob/4b755bec0484ff756f1eb51e9c0f57b5bd5cbdb3/pkg/cmd/project/shared/queries/export_data_test.go

iamazeem committed:
Update test

iamazeem committed:
Merge branch 'trunk' into 10301-gh-project-iteration-id

iamazeem commented:
@andyfeller:
Thank you for pointing that out!
I've updated the failing test.
Tests are now passing locally and in the CI.
LMK if there are other tests that need to be added/updated.
As I'm new to codebase, any pointers would greatly help. Thanks!

andyfeller committed:
Ensure item iteration field is populated in tests

Updating tests to ensure `ProjectV2ItemFieldIterationValue` actually populates `IterationId` appropriately.

andyfeller reviewed (approved):
LGTM following 5f54201fc5189fa41dfcb60bf6419b94fdbfa7ed fixes to ensure `IterationId` field is marshalled / unmarshalled as expected.
